app/app_main.c

```diff
@@ -1289,6 +1306,65 @@ static ACVP_RESULT app_sha_handler(ACVP_TEST_CASE *test_case)
     return ACVP_SUCCESS;
 }
 
+static ACVP_RESULT app_hmac_handler(ACVP_TEST_CASE *test_case)
+{
+    ACVP_HMAC_TC	*tc;
+    const EVP_MD	*md;
+    HMAC_CTX       hmac_ctx;
+    int msg_len;
+
+    if (!test_case) {
+        return ACVP_INVALID_ARG;
+    }
+
+    tc = test_case->tc.hmac;
+
+    //printf("%s: enter (tc_id=%d)\n", __FUNCTION__, tc->tc_id);
+
+    switch (tc->cipher) {
+    case ACVP_HMAC_SHA1:
+      md = EVP_sha1();
+      break;
+    case ACVP_HMAC_SHA2_224:
+      md = EVP_sha224();
+      break;
+    case ACVP_HMAC_SHA2_256:
+      md = EVP_sha256();
+      break;
+    case ACVP_HMAC_SHA2_384:
+      md = EVP_sha384();
+      break;
+    case ACVP_HMAC_SHA2_512:
+      md = EVP_sha512();
+      break;
+    default:
+    	printf("Error: Unsupported hash algorithm requested by ACVP server\n");
```

Comment:
Looks like you have the unsupported error I was looking for in the other comment - so disregard that :)